src/main/java/com/greenfox/dramacsoport/petclinicbackend/services/JwtService.java

```diff
@@ -47,6 +49,19 @@ public String extractUsername(String jwt) {
         return claims.getSubject();
     }
 
+    public Role extractRole(String jwt) {
+        Claims claims = getClaims(jwt);
+        String roleAsString = claims.get("role", String.class);
+        Role role;
+        switch (roleAsString) {
+            case "USER" -> role = Role.USER;
+            case "VET" -> role = Role.VET;
+            case "ADMIN" -> role = Role.ADMIN;
+            default -> throw new UnIdentifiableRoleException();
+        }
+        return role;
```

Comment:
```suggestion
        return switch (roleAsString) {
            case "USER" -> Role.USER;
            case "VET" -> Role.VET;
            case "ADMIN" -> Role.ADMIN;
            default -> throw new UnIdentifiableRoleException();
        }
```

Reply:
The fact that you are not returning as is and you are able to modify the role after the switch might be harmful.